File: nyayasahayak-main-main/src/core/types/bailReckoner.ts
export enum RiskLevel {
    LOW = "low",
    MEDIUM = "medium",
    HIGH = "high",
    CRITICAL = "critical"
}

export enum BailRecommendation {
    GRANT = "grant",
    GRANT_WITH_CONDITIONS = "grant_with_conditions",
    REJECT = "reject"
}

export interface RiskFactor {
    category: string;
    description: string;
    risk_level: RiskLevel;
    weight: number;
}

export interface BailReport {
    id: string;
    case_id: string;
    accused_id: string;
    accused_name: string;
    offenses: string[];
    is_bailable_offense: boolean;
    max_punishment_years: number;
    criminal_history_score: number;
    flight_risk_score: number;
    tampering_risk_score: number;
    overall_risk_score: number;
    risk_factors: RiskFactor[];
    recommendation: BailRecommendation;
    suggested_conditions: string[];
    created_at: string;
    metadata: Record<string, any>;
}

export interface BailAnalysisRequest {
    case_id: string;
    accused_id: string;
    offenses: string[];
    prior_convictions?: number;
    is_flight_risk?: boolean;
    is_witness_tampering_risk?: boolean;
}

File: backend/app/schemas/bail_reckoner.py
"""
Bail Reckoner Schemas - Skill 09
AI-driven bail eligibility analysis
"""
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime
from enum import Enum


class RiskLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class BailRecommendation(str, Enum):
    GRANT = "grant"
    GRANT_WITH_CONDITIONS = "grant_with_conditions"
    REJECT = "reject"


class RiskFactor(BaseModel):
    category: str  # Flight Risk, Tampering, Reoffending
    description: str
    risk_level: RiskLevel
    weight: float


class BailReport(BaseModel):
    id: str
    case_id: str
    accused_id: str
    accused_name: str
    
    offenses: List[str]  # e.g., ["BNS 303(2)", "BNS 318(4)"]
    is_bailable_offense: bool
    max_punishment_years: int
    
    # Analysis
    criminal_history_score: float
    flight_risk_score: float
    tampering_risk_score: float
    overall_risk_score: float  # 0-100
    
    risk_factors: List[RiskFactor]
    recommendation: BailRecommendation
    suggested_conditions: List[str]
    
    created_at: datetime
    metadata: Dict[str, Any] = {}


class BailAnalysisRequest(BaseModel):
    case_id: str
    accused_id: str
    offenses: List[str]
    prior_convictions: int = 0
    is_flight_risk: bool = False
    is_witness_tampering_risk: bool = False
